src/Symfony/Component/DependencyInjection/EnvVarProcessor.php

```diff
@@ -57,6 +57,8 @@ public static function getProvidedTypes(): array
             'enum' => \BackedEnum::class,
             'shuffle' => 'array',
             'defined' => 'bool',
+            'date_time' => \DateTime::class,
+            'date_time_immutable' => \DateTimeImmutable::class,
```

Comment:
`date_time_immutable` should be the default for `date_time`. No need to have 2 distinct processors. Allowing the mutable `DateTime` would lead to unexpected behaviors if it's used in 2 services with one of them altering the value.
```suggestion
            'date_time' => \DateTimeImmutable::class,
```

Reply:
I agree about that. I would not add support for the mutable value object.

Reply:
Done.